gcc/rust/backend/rust-compile-expr.cc

```diff
@@ -1163,6 +1163,36 @@ CompileExpr::array_copied_expr (Location expr_locus,
   unsigned HOST_WIDE_INT len
     = wi::ext (max - min + 1, precision, sign).to_uhwi ();
 
+#if 1
+  // TODO: Somehow check whether we are compiling in a const context, since we
+  //       obviously can't make a loop to init in such cases, and will need to
+  //       allocate/init the array (however large) at compile time...
```

Comment:
What would be the best way to check whether we are compiling an array in a `const` item?

Reply:
I assume it would probably be by using some form of context information: If we are compiling a constant assignment, `push_context(CONST)`, visit the initializer and then `pop_context()`. I've never touched this part of the codebase so unsure on how all of that maps! This is what we had to do for macro expansion, where you do want specific behavior based on the "parent" of the thing you're currently handling. In that case, you want information about the "parent" which will receive/contain the array you're compiling, in a way